sdk/daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/MetadataHashV1Spec.scala

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf
+
+import com.digitalasset.daml.lf.crypto.Hash
+import com.digitalasset.daml.lf.crypto.HashUtils.HashTracer
+import com.digitalasset.daml.lf.data.{Bytes, Ref, Time}
+import com.digitalasset.daml.lf.transaction.{FatContractInstance, NodeId, TransactionSpec}
+import com.digitalasset.daml.lf.value.Value
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+import java.time.Duration
+import java.util.UUID
+import scala.collection.immutable.{SortedMap, SortedSet}
+
+class MetadataHashV1Spec extends AnyWordSpec with Matchers with HashUtils {
+
+  private implicit val contractIdOrdering: Ordering[Value.ContractId] = Ordering.by(_.coid)
+  private implicit val nodeIdOrdering: Ordering[NodeId] = Ordering.by(_.index)
+  private val transactionUUID = UUID.fromString("4c6471d3-4e09-49dd-addf-6cd90e19c583")
+  private val cid1 = Value.ContractId.assertFromString(
+    "0007e7b5534931dfca8e1b485c105bae4e10808bd13ddc8e897f258015f9d921c5"
+  )
+  private val cid2 = Value.ContractId.assertFromString(
+    "0059b59ad7a6b6066e77b91ced54b8282f0e24e7089944685cb8f22f32fcbc4e1b"
+  )
+  private val alice = Ref.Party.assertFromString("alice")
+  private val bob = Ref.Party.assertFromString("bob")
+  private val node1 = FatContractInstance.fromCreateNode(
+    TransactionSpec.dummyCreateNode(cid1, Set(alice), Set(alice)),
+    Time.Timestamp.Epoch.add(Duration.ofDays(10)),
+    Bytes.assertFromString("0010"),
+  )
+  private val node2 = FatContractInstance.fromCreateNode(
+    TransactionSpec.dummyCreateNode(cid2, Set(bob), Set(bob)),
+    Time.Timestamp.Epoch.add(Duration.ofDays(20)),
+    Bytes.assertFromString("0050"),
+  )
+  private val metadata = Hash.TransactionMetadataBuilderV1.Metadata(
+    actAs = SortedSet(alice, bob),
+    commandId = Ref.CommandId.assertFromString("command-id"),
+    submissionId = Ref.SubmissionId.assertFromString("submission-id"),
+    transactionUUID = transactionUUID,
+    mediatorGroup = 0,
+    domainId = "domainId",
+    ledgerEffectiveTime = Some(Time.Timestamp.Epoch),
+    submissionTime = Time.Timestamp.Epoch,
+    nodeSeeds = SortedMap(
+      NodeId(1) -> Hash.assertFromString(
+        "d70605b7a7398f79c0aaa7a280ac0fa7ca079dce3f9a8f1f1d1044f82822591e"
+      ),
+      NodeId(2) -> Hash.assertFromString(
+        "9724b5d6e86d3d8a5e1682c56958c05a7785a5814e22edae37aaba54a8258031"
+      ),
+    ),
```

Comment:
Very minor (you may ignore it): It is maybe easier to debug with something like that.
```suggestion
      NodeId(1) -> Hash.assertFromString("01" * 32),
      NodeId(2) -> Hash.assertFromString("02" * 32)              ),
    ),
```